### ISSUE TITLE
Apply addition weights in the pre-compute 

### ISSUE THREAD

**Issue by dr-guangtou** [alexieleauthaud/DeltaSigmaPipelineHSC#42]
By default, the lens and random catalog will have a column named "weight".  This will be an additional weights for the lenses, right now they are all 1.0 as placeholder.  

We need to apply this weight during the pre-compute step.

**Comment by alexieleauthaud** [comment]
No, we can simply apply this weight in the final stage, right? Precompute doesn’t care about the lens weight, I’m pretty sure it can be applied after the fact.

**Comment by dr-guangtou** [comment]
@alexieleauthaud Yes, it can be applied in the computeDS step.  Then at least we should save this addition weights in the precompute results.  Otherwise, we have to load the lens and random catalog during the computeDS phase to have access to them. 

To select useful lenses, we are reading lenses catalog.  But I am thinking about the most complicated situation: when catalog 1 (e.g. lenses) and catalog 2 (e.g. randoms) both have additional weights to apply.  

I will save the weights for now.

**Comment by alexieleauthaud** [comment]
Don’t we have the read in the lens file anyway, to apply the lens selection (when you define which lens sample you want to compute DS for?)

**Comment by dr-guangtou** [comment]
Yes, we need to read lens, but not lens and random at the same time.  

Now we simply save the "weight" in the precompute results.
